server/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -219,12 +221,13 @@ private static Scheduler buildScheduler(ScheduledThreadPoolExecutor scheduledThr
 
     private final AtomicLong executionIdGen = new AtomicLong();
 
-    private BulkRequest bulkRequest;
+    private volatile BulkRequest bulkRequest;
```

Comment:
volatile should not be necessary here given that all access is done under lock

Reply:
fixed in https://github.com/elastic/elasticsearch/pull/41451/commits/a9e0f3bd8db3b7aa79ba2166895057c198475f56